R/linkMultipleVariants.R

```diff
@@ -0,0 +1,294 @@
+    for (are in argsRecEqual) {
+        for (parm in paramsSeparate) {
+            if (!identical(parm[[are]], combinedDigestParams[[are]])) {
+                message("We recommend that the argument ", are,
+                        " is set to the same value in the ",
+                        "combined and all the separate runs")
+            }
+        }
+    }
+
+    ## Adding UMI counts after the processing may not be accurate - will
+    ## always use read counts
+    if (length(grep("U", paste0(combinedDigestParams$elementLengthsForward,
+                                combinedDigestParams$elementLengthsReverse))) > 0) {
+        warning("Aggregating UMI counts may not be accurate - will use ",
+                "read counts instead")
+    }
+
+    ## Get number and lengths of forward and reverse variable sequences
+    nbrFwdV <- sum(strsplit(combinedDigestParams$elementsForward, "")[[1]] == "V")
+    nbrRevV <- sum(strsplit(combinedDigestParams$elementsReverse, "")[[1]] == "V")
+    ## Check each of the separate runs
+    lengthsV <- unlist(lapply(paramsSeparate, function(parm) {
+        if (!parm$mergeForwardReverse) {
+            ## Don't merge - there can be only one V in total, and the length
+            ## must be given
+            els <- c(strsplit(parm$elementsForward, "")[[1]],
+                     strsplit(parm$elementsReverse, "")[[1]])
+            ell <- c(parm$elementLengthsForward, parm$elementLengthsReverse)
+            if (sum(els == "V") != 1) {
+                stop("Must have exactly one V in each separate run")
+            }
+            if (any(ell[which(els == "V")] == -1)) {
+                stop("The length of the variable sequence must be explicitly ",
+                     "specified in the separate runs")
+            }
+            ell[which(els == "V")]
+        } else {
+            ## Merge - there must be a V in both fwd and reverse, with the
+            ## same length
+            elsfwd <- strsplit(parm$elementsForward, "")[[1]]
+            elsrev <- strsplit(parm$elementsReverse, "")[[1]]
+            ellfwd <- parm$elementLengthsForward
+            ellrev <- parm$elementLengthsReverse
+            indexVfwd <- which(elsfwd == "V")
+            indexVrev <- which(elsrev == "V")
+            if (length(indexVfwd) != 1 || length(indexVrev) != 1) {
+                stop("Must have exactly one V in the forward and one V in ",
+                     "the reverse read in each separate run for ",
+                     "mergeForwardReverse=TRUE")
+            }
+            if ((ellfwd[indexVfwd] != ellrev[indexVrev]) ||
+                any(c(ellfwd[indexVfwd], ellrev[indexVrev]) == -1)) {
+                stop("The length of the variable sequence must be explicitly ",
+                     "specified and identical between the forward and reverse ",
+                     "reads in the separate runs for mergeForwardReverse=TRUE")
+            }
+            ellfwd[indexVfwd] + ellrev[indexVrev]
```

Comment:
I have introduced `indexVfwd` and `indexVrev` here - hopefully making it a bit simpler to read.
Also, I have added a "return value" for the `lapply` (line 186), I hope that's correct

Reply:
Yes, that's more straightforward. I think the return value should be only `ellfwd[indexVfwd]` (since the fwd and rev var seqs are merged, the total length is the same as the individual ones.

Reply:
Right, I corrected the return value to be `ellfwd[indexVfwd]` in 5a42428